lib/shared/bucketing-assembly-script/assembly/eventQueue/requestPayloadManager.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    DVCEvent,
+    DVCUser,
+    DVCPopulatedUser,
+    DVCRequestEvent,
+    FlushPayload,
+    UserEventsBatchRecord, EventQueueOptions
+} from '../types'
+import { JSON } from 'assemblyscript-json/assembly'
+import { _getPlatformData } from '../managers/platformDataManager'
+import { AggEventQueue } from './eventQueue'
+
+/**
+ * This RequestPayloadManager Class handles all the logic for creating event flushing payloads,
+ * and handling the onSuccess and onFailure logic for those payloads.
+ */
+export class RequestPayloadManager {
+    private pendingPayloads: Map<string, FlushPayload>
+    private readonly chunkSize: i32
+
+    constructor(options: EventQueueOptions) {
+        this.pendingPayloads = new Map<string, FlushPayload>()
+        this.chunkSize = options.eventRequestChunkSize
+    }
+
+    constructFlushPayloads(
+        userEventQueue: Map<string, UserEventsBatchRecord>,
+        aggEventQueue: AggEventQueue
+    ): FlushPayload[] {
+        this.checkForFailedPayloads()
+
+        // Add events from user event queue
+        const userEventQueueKeys = userEventQueue.keys()
+        for (let i = 0; i < userEventQueueKeys.length; i++) {
+            const key = userEventQueueKeys[i]
+            const userEventsRecord = userEventQueue.get(key)
+            this.addEventsToPendingPayloads(userEventsRecord)
+        }
+
+        // Add events from aggregate events
+        this.addAggEventsToPendingPayloads(aggEventQueue)
+
+        return this.pendingPayloads.values()
+    }
+
+    /**
+     * generate aggregated events by resolving aggregated event map into DVCEvent's.
+     */
+    private addAggEventsToPendingPayloads(
+        aggEventQueue: AggEventQueue
+    ): void {
+        const aggEventQueueKeys = aggEventQueue.keys()
```

Comment:
it would be a lot easier to understand this part if the various `keys` arrays were named after what they actually contain, like `featureIds` and whatnot